### ISSUE TITLE
What Alpha socket is being use to determine Alphamode - always defaults to BLEND (should be OPAQUE)?

### ISSUE THREAD

**Issue by ronh991** [KhronosGroup/glTF-Blender-IO#2261]
**Describe the bug**
Depending on the shader node set up, my export will always default to alphaMode = BLEND , this cause flickering in the MSFS game engine. as the materials are always trying to blend with their background textures etc.

the gather_alpha_info function in gltf2_blender_search_node_tree.py will always pass through to the last if and default to BLEND

```
    # Set alpha mode
    if info['alphaMode'] is None:
        # Is zero? Weird, but okay.
        if info['alphaFactor'] == 0:
            info['alphaMode'] = 'MASK'
            info['alphaCutoff'] = 0.5
        else:
            info['alphaMode'] = 'BLEND'

```

Blender BSDF has alphamode naming as straight, premultiplied, channel packed and none.  Is there a conflict with BLEND, OPAQUE, DITHER and MASK

**To Reproduce**
Utilize a shader node structure where alpha from texture is not connected in the base color, and utilize the ASOBO exporter updated to 4.2 (with all the other issues resolved)

**Expected behavior**
The default alphamode should be OPAQUE?

**Screenshots**

**.blend file/ .gltf (mandatory)**
a blend file will not work without the MSFS exporter 
edit: yes it does set to blend without the  MSFS/ASOBO exporter

[alphamode_test.zip](https://github.com/user-attachments/files/15571517/alphamode_test.zip)

**Version**
 - OS: win 10
 - Blender Version 4.2 Jun 3 b375

**Additional context**
trying to understand where the to_node_nav selects the alpha socket.

an example of error info is these materials should be opaque

```
		{
			"alphaMode":"BLEND",
			"name":"Tailwind Mechanicals",
			"pbrMetallicRoughness":{
				"baseColorTexture":{
					"index":3
				},
				"metallicFactor":0.1090909093618393,
				"roughnessFactor":0.2672727108001709
			}
		},
		{
			"alphaMode":"BLEND",
			"name":"Yellow Plastic",
			"pbrMetallicRoughness":{
				"baseColorFactor":[
					0.6104918718338013,
					0.7991039156913757,
					0.07818762212991714,
					0.3436855673789978
				],
				"metallicFactor":0,
				"roughnessFactor":0.30000001192092896
			}
		},

```

however this material has to have blend - windshield transparent

```
		{
			"alphaMode":"BLEND",
			"name":"Tailwind Mechanicals",
			"pbrMetallicRoughness":{
				"baseColorTexture":{
					"index":0
				},
				"metallicFactor":0.1090909093618393,
				"roughnessFactor":0.2672727108001709
			}
		}

```

**Comment by ronh991** [comment]
Thank you, my point being - this is new functionality to me.  The April 6th branch of 4.2 and 4.1.1 respects my setting the alphamode to OPAQUE.  This is a game engine restriction - I can't change.

4.1 does this

``
		{
			"name":"Fuselage",
			"normalTexture":{
				"index":0
			},
			"occlusionTexture":{
				"index":1
			},
			"pbrMetallicRoughness":{
				"baseColorTexture":{
					"index":2
				},
				"metallicFactor":0,
				"metallicRoughnessTexture":{
					"index":1
				},
				"roughnessFactor":0.5
			}
		},
```


4.2 now does this

```
		{
			"alphaMode":"BLEND",
			"name":"Fuselage",
			"normalTexture":{
				"index":0
			},
			"occlusionTexture":{
				"index":1
			},
			"pbrMetallicRoughness":{
				"baseColorTexture":{
					"index":2
				},
				"metallicFactor":0,
				"metallicRoughnessTexture":{
					"index":1
				},
				"roughnessFactor":0.5
			}
		},

```


As to the Straight, pre multipled - that is an image.alpha_mode variable in Blender. - so I asked if this is new or why it has the same name as alphamode. 

Seems I am getting things mixed up as the naming does not match the my expectations. Blend Mode is OPAQUE, BLEND etc. but the gltf calls it alphaMode

**Comment by julienduroure** [comment]
Starting Blender 4.2 beta (today), this option does not exist anymore in Blender

![image](https://github.com/KhronosGroup/glTF-Blender-IO/assets/1696020/86c2172b-c939-444b-b8dd-b6172e1df02a)

So, alphaMode can't be set anymore with this no more existing Blender value.
We now have to analyze the node tree to detect what alphamode should be exported:

- Alpha == 1 => OPAQUE
- Using a special node setup => MASK
- Else => BLEND

**Comment by ronh991** [comment]
Ack! - ok will try to work around this with material_hook.

But there is still a alphaMode gltf json variable right?

**Comment by julienduroure** [comment]
glTF 2.0 specification didn't change

**Comment by ronh991** [comment]
Using material hook - I was able to fix - Only works with 4.2 Beta release 01b8
